driver-core/src/main/java/com/datastax/driver/core/utils/LongAdderFactory.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.utils;
+
+import com.google.common.base.Throwables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A factory for {@link LongAdder} instances.
+ * <p>
+ * This factory will pick the best implementation based on what is available at runtime:
+ * <ol>
+ * <li>If JDK 8's {@code java.util.concurrent.atomic.LongAdder} is present, then an implementation
+ * delegating to that class will be used;</li>
+ * <li>If {@code sun.misc.Unsafe} is present, then an implementation
+ * using that class will be used (this implementation is actually very similar to the original
+ * {@code java.util.concurrent.atomic.LongAdder} class);</li>
+ * <li>Otherwise, a default implementation will be used. Beware that this implementation might have
+ * worse performances that the ones mentioned above.</li>
+ * </ol>
+ */
+public final class LongAdderFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LongAdderFactory.class);
+
+    private static final String JDK8_LONG_ADDER_CLASS_NAME = "com.datastax.driver.core.utils.Jdk8LongAdder";
+
+    private static final String UNSAFE_LONG_ADDER_CLASS_NAME = "com.datastax.driver.core.utils.UnsafeLongAdder";
+
+    private static final boolean JDK8_LONG_ADDER_AVAILABLE;
+
+    private static final boolean UNSAFE_LONG_ADDER_AVAILABLE;
+
+    static {
+        JDK8_LONG_ADDER_AVAILABLE = isClassAvailable(JDK8_LONG_ADDER_CLASS_NAME);
```

Comment:
Since the test actually tries to instantiate our classes, it boils down to the same, e.g. if `java.util.concurrent.atomic` is not available then `com.datastax.driver.core.utils.Jdk8LongAdder` cannot be instantiated.